File: examen_backend/lib/examen_backend/hotels/hotel.ex
defmodule ExamenBackend.Hotels.Hotel do
  @moduledoc false
  use Ecto.Schema

  import Ecto.Changeset

  alias ExamenBackend.Hotels.HotelRoom

  @primary_key {:id, :binary_id, autogenerate: true}
  schema "hotels" do
    field :name, :string
    has_many :hotel_rooms, HotelRoom, on_delete: :delete_all

    timestamps()
  end

  @doc false
  def changeset(hotel, attrs, _opts \\ []) do
    hotel
    |> cast(attrs, [:name])
    |> validate_required([:name])
  end

end

File: examen_frontend/src/Interfaces/Hotel.ts
export interface HotelBase {
    name: string;
  }
  
  export interface Hotel extends HotelBase {
    hotel_rooms?: HotelRoom[];
    id: number;
  }
  
  export interface NewHotel extends HotelBase {
    id?: undefined;
  }

  export interface HotelRoomBase {
    name: string;
    size: string;
  }

  export interface HotelRoom extends HotelRoomBase {
    id: number;
  }

  export interface NewHotelRoom extends HotelRoomBase {
    id?: undefined;
  }